src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -983,11 +994,54 @@ internal override IOperation GetOperationWorker(CSharpSyntaxNode node, GetOperat
                     break;
             }
 
-            // The CSharp operation factory assumes that UnboundLambda will be bound for error recovery and never be passed to the factory
-            // as the start of a tree to get operations for. This is guaranteed by the builder that populates the node map, as it will call
-            // UnboundLambda.BindForErrorRecovery() when it encounters an UnboundLambda node.
-            Debug.Assert(result.Kind != BoundKind.UnboundLambda);
-            return _operationFactory.Create(result);
+            return result;
+        }
+
+        internal override IOperation GetOperationWorker(MethodSymbol method, CSharpSyntaxNode bodySyntax, CancellationToken cancellationToken)
+        {
+            IOperation result = null;
+            var node = GetBoundNode(bodySyntax, GetOperationOptions.Highest);
+
+            if (!node.HasErrors)
+            {
+                var body = (BoundStatement)node;
+                var loweredBody = LowerMethodBody(method, body);
+
+                // The CSharp operation factory assumes that UnboundLambda will be bound for error recovery and never be passed to the factory
+                // as the start of a tree to get operations for. This is guaranteed by the builder that populates the node map, as it will call
+                // UnboundLambda.BindForErrorRecovery() when it encounters an UnboundLambda node.
+                Debug.Assert(loweredBody.Kind != BoundKind.UnboundLambda);
+                result = _operationFactory.Create(loweredBody);
```

Comment:
>result = _operationFactory.Create(loweredBody); [](start = 16, length = 47)

It feels wrong to tie lowered tree to any SemanticModel. SemanticModel deals with un-lowered bound trees, who knows what negative effects will this operation have, how will it affect the caches, etc. I think there should be no interaction between SemanticModel and the tree built of off lowered bound nodes.